### PR TITLE
Regression fix: bootstrap confirm modal

### LINKED ISSUE #4324: Unable to delete widget
Hello ,
This is what am getting when i try to delete widget from a page.
Capture

Any idea on how to resolve this ?
TargetFramework netcoreapp3.0 TargetFramework

PackageReference Include="OrchardCore.Application.Cms.Targets" Version="1.0.0-beta3-72585" 

PackageReference Include="OrchardCore.Application.Targets" Version="1.0.0-beta3-72585"

### PR DESCRIPTION
**Pull request by hishamco**
Fix #4324

### CODE REVIEW

**Comment by hishamco** [comment]
@jptissot the issue got fixed, I will try to open another PR to fix it everywhere, this happen after that latest change in bootstrap confirm modal (adding custom css classes)

File: src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ $(function () {
     $(document).on('click', '.widget-delete', function () {
         var title = $(this).data('title');
         var message = $(this).data('message');
-        confirmDialog(title, message, r => {
+        confirmDialog(title, message, undefined, undefined, undefined, undefined, r => {
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js]: Maybe an improvement would be to pass a settings object to the `confirmDialog`. function
```
var options = {title: 'title', message: 'message', callback: function(x){} ...};
confirmDialog(options);
```

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js]: Perhaps I was running on a slow internet connection, if you look closely I just created a new branch few minutes ago, to improve this by using optional params instead

Comment (jptissot) [reply src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js]: Make sure to search and fix all calls of the confirmDialog() function

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js]: Sure, they are in three modules only

**Comment by hishamco** [comment]
Just I need to rebase on dev and run `gulp build`

**Comment by hishamco** [comment]
Just I need to run `gulp build`

**Comment by jptissot** [comment]
I think you pushed to dev instead of your branch.

File: src/OrchardCore.Themes/TheAdmin/Assets/js/admin.js [hunk 2]
```diff
@@ -1,27 +1,10 @@
...
-
-    if (okCssClass === undefined) {
-        okCssClass = $('#confirmRemoveModalMetadata').data('okClass');
-    }
-
-    if (cancelCssClass === undefined) {
-        cancelCssClass = $('#confirmRemoveModalMetadata').data('cancelClass');
-    }
+function confirmDialog({title, message, okText, cancelText, okCssClass, cancelCssClass, callback}) {
+    title = title || $('#confirmRemoveModalMetadata').data('title');
```
Comment (jptissot) [inline src/OrchardCore.Themes/TheAdmin/Assets/js/admin.js]: repeating $('#confirmRemoveModalMetadata') is not good for perf

File: src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js [hunk 3]
```diff
@@ -58,12 +58,12 @@ $(function () {
     $(document).on('click', '.widget-delete', function () {
         var title = $(this).data('title');
         var message = $(this).data('message');
-        confirmDialog(title, message, r => {
+        confirmDialog({title: title, message: message, callback: r => {
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js]: arrow not supported in IE

File: src/OrchardCore.Modules/OrchardCore.Contents/Views/Admin/List.cshtml [hunk 4]
```diff
@@ -149,15 +149,12 @@
 
         $(".dropdown-menu .dropdown-item").on("click", function () {
             if ($(":checkbox[name='itemIds']:checked").length > 1) {
-
-                var title = $(this).data('title');
-                var message = $(this).data('message');
-                confirmDialog(title, message, r => {
+                confirmDialog({title: $(this).data('title'), message: $(this).data('message'), callback: r => {
```
Comment (jptissot) [inline src/OrchardCore.Modules/OrchardCore.Contents/Views/Admin/List.cshtml]: arrow function are not supported in IE https://caniuse.com/#feat=arrow-functions

File: src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js (hunk 3 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Flows/wwwroot/Scripts/flows.edit.js]: Seems we fixed in the admin before

**Comment by jptissot** [comment]
I will push a commit to the branch if that is ok

**Comment by hishamco** [comment]
Feel free to push without asking :)

**Comment by jptissot** [comment]
I am pushing a commit to this and testing. Will let you know when it's ready.

**Comment by jptissot** [comment]
I think this now fixes most issues with modals

**Comment by hishamco** [comment]
Ya

**Comment by jptissot** [comment]
@deanmarcussen I tested a lot of places where the modal is used and it works great.

**Comment by jptissot** [comment]
Strange about the version of popper.js. All I did was npm install + npm run build. Looking at the lock file, it seems that there is a dependency that uses vers 1.15 which somehow got installed ?

**Comment by jptissot** [comment]
https://github.com/OrchardCMS/OrchardCore/blob/dev/package-lock.json#L5283

**Pull request merged**